File: zhuzhasi/ДЗ (8)/cpp/task1.cpp
#include <iostream>
#include <string>

class Student {
protected:
    std::string name;
    std::string student_id;

public:
    Student(const std::string& name, const std::string& student_id)
        : name(name), student_id(student_id) {}

    virtual void study() const {
        std::cout << "Студент учится" << std::endl;
    }

    virtual ~Student() = default;
};

class Undergraduate : public Student {
public:
    Undergraduate(const std::string& name, const std::string& student_id)
        : Student(name, student_id) {}

    void study() const override {
        std::cout << "Бакалавр усердно учится" << std::endl;
    }
};

class Graduate : public Student {
public:
    Graduate(const std::string& name, const std::string& student_id)
        : Student(name, student_id) {}

    void study() const override {
        std::cout << "Магистр углубляется в исследования" << std::endl;
    }
};

int main() {
    Undergraduate u("Иван", "U001");
    Graduate g("Мария", "G001");

    u.study();
    g.study();

    return 0;
}

File: zhuzhasi/ДЗ (8)/cs/task1.cs
using System;

class Student
{
    protected string name;
    protected string studentId;

    public Student(string name, string studentId)
    {
        this.name = name;
        this.studentId = studentId;
    }

    public virtual void Study()
    {
        Console.WriteLine("Студент учится");
    }
}

class Undergraduate : Student
{
    public Undergraduate(string name, string studentId) : base(name, studentId) {}

    public override void Study()
    {
        Console.WriteLine("Бакалавр усердно учится");
    }
}

class Graduate : Student
{
    public Graduate(string name, string studentId) : base(name, studentId) {}

    public override void Study()
    {
        Console.WriteLine("Магистр углубляется в исследования");
    }
}

class Task1
{
    static void Main()
    {
        Undergraduate u = new Undergraduate("Иван", "U001");
        Graduate g = new Graduate("Мария", "G001");

        u.Study();
        g.Study();
    }
}
